python/cucim/src/cucim/core/operations/intensity/kernel/cuda_kernel_source.py

```diff
@@ -169,74 +169,71 @@
     output_tensor[(blockIdx.z * pitch) +
                   ((out_pixel_h + out_h_start) * input_w) +
                   (out_pixel_w + out_w_start)] = sum_;
+  }
+}
 
-    // replicate along top edge
-    if (out_pixel_h == 0) {
-      for (int ik = 0; ik < out_h_start; ik++)
-        output_tensor[(blockIdx.z * pitch) +
-                      ((out_pixel_h + ik) * input_w) +
-                      (out_pixel_w + out_w_start)] = sum_;
-    }
-
-    // replicate along bottom edge
-    if (out_pixel_h == (output_h - 1)) {
-      for (int ik = 1; ik <= out_h_end; ik++)
-        output_tensor[(blockIdx.z * pitch) +
-                      ((out_h_start + out_pixel_h + ik) * input_w) +
-                      (out_pixel_w + out_w_start)] = sum_;
-    }
-
-    // replicate along left edge
-    if (out_pixel_w == 0) {
-      for (int ik = 0; ik < out_w_start; ik++)
-        output_tensor[(blockIdx.z * pitch) +
-                      ((out_pixel_h + out_h_start) * input_w) + ik] = sum_;
-    }
-
-    // replicate along right edge
-    if (out_pixel_w == (output_w - 1)) {
-      for (int ik = 1; ik <= out_w_end; ik++)
-        output_tensor[(blockIdx.z * pitch) +
-                      ((out_pixel_h + out_h_start) * input_w) +
-                      (out_pixel_w + out_w_start + ik)] = sum_;
-    }
+__global__ void zoomout_edge_pad(float *output_tensor, int height, int width,
+                                  int pitch, int no_padding_h_start,
+                                  int no_padding_w_start,
+                                  int no_padding_h_end, int no_padding_w_end) {
+  // H -> block Y, row
+  // W -> block X, col
 
-    // corner replication not very friendly if large area to patch -
-    // single thread issues stores
-    // ToDo: Consider adding another kernel for corner padding
+  int out_pixel_h = blockIdx.y * blockDim.y + threadIdx.y;
+  int out_pixel_w = blockIdx.x * blockDim.x + threadIdx.x;
+  int image_start_offset = blockIdx.z * pitch;
```

Comment:
can either remove this `image_start_offset` or use it below

Reply:
Removed.